tests/pytests/functional/formulas/test_docker.py

```diff
@@ -0,0 +1,82 @@
+        "cmd_|-docker-software-docker-archive-install-managed-service_|-systemctl daemon-reload_|-run",
+        "file_|-docker-compose-software-binary-install-symlink-docker-compose_|-/usr/local/bin/docker-compose_|-symlink",
+        "file_|-docker-compose-software-binary-install_|-/usr/local/docker-compose-latest/bin//docker-compose_|-managed",
+        "file_|-docker-software-daemon-file-managed-daemon_file_|-/etc/docker/daemon.json_|-absent",
+        "file_|-docker-software-docker-archive-install-file-directory_|-/var/lib/docker_|-directory",
+        "file_|-docker-software-docker-archive-install-managed-service_|-/usr/lib/systemd/system/docker.service_|-managed",
+        "file_|-docker-software-docker-archive-install-symlink-containerd-shim_|-/usr/local/bin/containerd-shim_|-symlink",
+        "file_|-docker-software-docker-archive-install-symlink-containerd_|-/usr/local/bin/containerd_|-symlink",
+        "file_|-docker-software-docker-archive-install-symlink-ctr_|-/usr/local/bin/ctr_|-symlink",
+        "file_|-docker-software-docker-archive-install-symlink-docker-init_|-/usr/local/bin/docker-init_|-symlink",
+        "file_|-docker-software-docker-archive-install-symlink-docker-proxy_|-/usr/local/bin/docker-proxy_|-symlink",
+        "file_|-docker-software-docker-archive-install-symlink-docker_|-/usr/local/bin/docker_|-symlink",
+        "file_|-docker-software-docker-archive-install-symlink-dockerd_|-/usr/local/bin/dockerd_|-symlink",
+        "file_|-docker-software-docker-archive-install-symlink-runc_|-/usr/local/bin/runc_|-symlink",
+        "file_|-docker-software-docker-archive-install_|-/usr/local/docker-19.03.9/bin/_|-directory",
+        "pkg_|-docker-compose-software-binary-install_|-python-docker_|-installed",
+        "pkg_|-docker-compose-software-binary-install_|-python-pip_|-installed",
+        "pkg_|-docker-software-docker-archive-install_|-python-docker_|-installed",
+        "pkg_|-docker-software-docker-archive-install_|-python-pip_|-installed",
+        "service_|-docker-software-service-running-docker_|-docker_|-running",
+        "service_|-docker-software-service-running-unmasked_|-docker_|-unmasked",
+        "service_|-docker-software-service-running-docker-fail-notify_|-docker_|-enabled",
+        "test_|-docker-compose-package-install-other_|-docker-compose-package-install-other_|-show_notification",
+        "test_|-docker-software-desktop-install-other_|-docker-software-desktop-install-other_|-show_notification",
+        "test_|-docker-software-package-install-other_|-docker-software-package-install-other_|-show_notification",
+    ]
+    for state_id in state_ids:
+        assert ret["local"][state_id]["result"] is not False
+
+    state_ids = [
+        "test_|-docker-software-service-running-docker-fail-notify_|-docker-software-service-running-docker-fail-notify_|-fail_without_changes",
+    ]
+    for state_id in state_ids:
+        assert ret["local"][state_id]["result"] is False
```

Comment:
```suggestion
def test_docker_formula(modules):
    ret = modules.state.sls("docker", test=True)
    for staterun in ret:
        if staterun.state_id == "test_|-docker-software-service-running-docker-fail-notify_|-docker-software-service-running-docker-fail-notify_|-fail_without_changes":
            assert staterun.result is False
        else:
            assert staterun.result is True
```
The returned object will be [MultiStateResult](https://github.com/saltstack/pytest-salt-factories/blob/1.0.0rc21/src/saltfactories/utils/functional.py#L463) for `state.sls` and each entry will be an instance of [`StateResult`](https://github.com/saltstack/pytest-salt-factories/blob/1.0.0rc21/src/saltfactories/utils/functional.py#L329)

Reply:
This way we're not going out to a shell, which we shouldn't, unless that's the purpose of the test.

Reply:
Of course, this doesn't just apply to this test module, but all the ones you provided.